yellowbrick/features/rankd.py

```diff
@@ -240,27 +275,173 @@ def rank(self, X, algorithm=None):
+                 colormap='RdBu_r', **kwargs):
+        """
+        Initialize the class with the options required to rank and
+        order features as well as visualize the result.
+        """
+        super(Rank1D, self).__init__(ax=None, algorithm=algorithm, features=features,
+                         colormap=colormap, **kwargs)
+
+    def draw(self, R, **kwargs):
+        """
+        Draws the heatmap of the ranking matrix of variables.
+        """
+        self.ax.bar(np.arange(len(R)), R, color=self.colormap[0])
+        self.ax.set_xticks(np.arange(len(R)))
+        self.ax.xaxis.grid(False)
+        self.ax.set_xticklabels(self.features_, rotation=90, ha='left')
+
+
+##########################################################################
+## Rank 2D Feature Visualizer
+##########################################################################
+
+class Rank2D(RankDBase):
+    """
+    Rank2D performs pairwise comparisons of each feature in the data set with
+    a specific metric or algorithm (e.g. Pearson correlation) then returns
+    them ranked as a lower left triangle diagram.
+
+    Parameters
+    ----------
+    ax : matplotlib Axes, default: None
+        The axis to plot the figure on. If None is passed in the current axes
+        will be used (or generated if required).
+
+    algorithm : one of {'pearson', 'covariance'}, default: 'pearson'
+        The ranking algorithm to use, default is Pearson correlation.
+
+    features : list
+        a list of feature names to use
+        If a DataFrame is passed to fit and features is None, feature
+        names are selected as the columns of the DataFrame.
+
+    colormap : string or cmap, default: 'RdBu_r'
+        optional string or matplotlib cmap to colorize lines
+        Use either color to colorize the lines on a per class basis or
+        colormap to color them on a continuous scale.
+
+    kwargs : dict
+        Keyword arguments that are passed to the base class and may influence
+        the visualization as defined in other Visualizers.
+
+    Examples
+    --------
+
+    >>> visualizer = Rank2D()
+    >>> visualizer.fit(X, y)
+    >>> visualizer.transform(X)
+    >>> visualizer.poof()
+
+    Notes
+    -----
+    These parameters can be influenced later on in the visualization
+    process, but can and should be set as early as possible.
+    """
+
+    ranking_methods = {
+        'pearson': lambda X: np.corrcoef(X.transpose()),
+        'covariance': lambda X: np.cov(X.transpose()),
+    }
+
+    def __init__(self, ax=None, algorithm='pearson', features=None,
+                 colormap='RdBu_r', **kwargs):
+        """
+        Initialize the class with the options required to rank and
+        order features as well as visualize the result.
+        """
+        super(Rank2D, self).__init__(ax=None, algorithm=algorithm, features=features,
+                         colormap=colormap, **kwargs)
+
+    def draw(self, R, **kwargs):
```

Comment:
RE: discussion from above; if you wouldn't mind adding docstring info about what `R` is in this case! 

Alternatively, we could remove the R argument to the method, and instead store it in `self.scores_` -- this would make the scores available to the user, which they might appreciate. What do you think about that?